core/src/apps/cardano/sign_tx.py

```diff
@@ -226,11 +227,39 @@ def _validate_outputs(
             )
 
         _validate_token_bundle(output.token_bundle)
+        _validate_max_tx_output_size(keychain, output, protocol_magic, network_id)
 
     if total_amount > LOVELACE_MAX_SUPPLY:
         raise wire.ProcessError("Total transaction amount is out of range!")
 
 
+def _validate_max_tx_output_size(
+    keychain: seed.Keychain,
+    output: CardanoTxOutputType,
+    protocol_magic: int,
+    network_id: int,
+) -> None:
+    """
+    Output size is currently limited to 4000 bytes at protocol level. Given
```

Comment:
I double checked the ledger specs and the memory limit apparently applies only to the "value" part of the utxo (i.e. not the address)

sources:

Ledger specs (notice "value" field is being extracted):
https://github.com/input-output-hk/cardano-ledger-specs/pull/2099/files#diff-1dbf0b3dcad4b1ddcdf86e9df16469b[…]461f3e9f906cf42d925d6fb3a386dd6c2R258

CDDL
https://github.com/input-output-hk/cardano-ledger-specs/blob/5e5b9a20f1b4818681996[…]e3859de49/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl

I also asked here for the value of the parameter for Alonzo era: https://github.com/input-output-hk/cardano-node/pull/2634#issuecomment-831111987

Reply:
Done https://github.com/vacuumlabs/trezor-firmware/pull/60/commits/a09056ccb5056b0ae4ed10ee7130ec53441c1d73. The actual limit can be adjusted later when the parameter value in Alonzo will be known.